publishing/models/envelope.py

```diff
@@ -256,28 +256,33 @@ class Meta:
 
     @classmethod
     def next_envelope_id(cls) -> str:
-        """Get packaged workbaskets where proc state SUCCESS."""
-        envelope = Envelope.objects.last_envelope_for_year()
+        """
+        Get packaged workbaskets where proc state SUCCESS.
 
-        if envelope is None:
-            # First envelope of the year.
-            now = datetime.today()
-            counter = max(1, int(settings.HMRC_PACKAGING_SEED_ENVELOPE_ID))
+        Provide the next envelope_id for regular publishing, manually publishing
+        and publishing in the new year.
+        """
+        # last envelope for the current year
+        previous_envelope = Envelope.objects.last_envelope_for_year()
+        seed_id = settings.HMRC_PACKAGING_SEED_ENVELOPE_ID
+        current_year = str(datetime.now().year)[-2:]
+
+        if (previous_envelope is None) and (seed_id[:2] != current_year):
+            # First envelope of the year in the format of YYCCCC.
+            counter = current_year + "0001"
         else:
-            year = int(envelope.envelope_id[:2])
             counter = max(
-                int(envelope.envelope_id[2:]) + 1,
-                int(settings.HMRC_PACKAGING_SEED_ENVELOPE_ID),
+                int(previous_envelope.envelope_id or 0) + 1,
```

Comment:
What's being guarded against by the `previous_envelope.envelope_id or 0`? I'm not sure what circumstances would result in `previous_envelope` being truthy / valid, but it's `envelope_id` being falsy.

Reply:
Interesting.... I had an edge case while building, but I can no longer replicate it so this is removed

Reply:
... but could `previous_envelope` itself be `None` / falsy?

Reply:
Somethig like this perhaps?
```suggestion
                (int(previous_envelope.envelope_id) + 1) if previous_envelope else 1
```